src/repl.ts

```diff
@@ -239,18 +308,48 @@ function _eval(service: Service, state: EvalState, input: string) {
     state.output = output;
   }
 
-  return changes.reduce((result, change) => {
-    return change.added ? exec(change.value, state.path) : result;
-  }, undefined);
+  let commands: Array<{ mustAwait?: true; caller: () => any }> = [];
+  let containsTLA = false;
+
+  for (const change of changes) {
+    if (change.added) {
+      if (
+        /\bawait\b/.test(change.value) &&
+        service.options.experimentalReplAwait
+      ) {
+        if (processTopLevelAwait === undefined) {
+          ({ processTopLevelAwait } = require('../dist-raw/node-repl-await'));
+        }
+
+        // Neline prevents comments to mess with wrapper
+        const wrappedResult = processTopLevelAwait(change.value + '\n');
+        if (wrappedResult !== null) {
+          containsTLA = true;
+          commands.push({
+            mustAwait: true,
+            caller: () => exec(wrappedResult, state.path, context),
+          });
+          continue;
+        }
+      }
+      commands.push({ caller: () => exec(change.value, state.path, context) });
```

Comment:
If the first command is async and has side-effects that affect the second command, then I believe the first command must be awaited before the second can start.

Reply:
I think it will also simplify things.  `_eval` will not need to return an array of commands, and callers will not
need to iterate them.  `_eval` can return a single value which might be a `Promise`.

Reply:
I think it will also simplify things.  `_eval` will not need to return an array of commands, and callers will not
need to iterate them.  `_eval` can return a single value which might be a `Promise`.

Reply:
I spoke too soon; I see that `caller()` wraps the `exec` call.

Reply:
Yes, that's why I return an array in first place, consumer is the one that end up executing each command.

Reply:
Also, just to clarify on this:
> I think it will also simplify things. _eval will not need to return an array of commands, and callers will not
need to iterate them. _eval can return a single value which might be a Promise.

_eval cannot return a promise, sync code and async code must be kept separately. As soon there's one promise involved for code that is sync only, tests will begin to fail since output will be returned in the wrong order. That's the reason why I did implement as it's now, to only resort to async usage when it's absolutely needed.